test/functional/runtime/autoload/provider/python3_spec.lua

```diff
@@ -0,0 +1,93 @@
+do
+  local proc =
+    io.popen([[python3 -c 'import neovim, sys; sys.stdout.write("ok")' 2> /dev/null]])
+  if proc:read() ~= 'ok' then
+    -- Don't run these tests if python3 is not available
+    return
+  end
+end
+
+
+local helpers = require('test.functional.helpers')
+local eval, command, feed = helpers.eval, helpers.command, helpers.feed
+local eq, clear, insert = helpers.eq, helpers.clear, helpers.insert
+local expect = helpers.expect
+
+
+describe('python3 commands and functions', function()
+  before_each(function()
+    clear()
+    command('python3 import vim')
+  end)
+
+  describe('feature test', function()
```

Comment:
these nested `describe()` blocks are not necessary in this case. Just name the `it()` block with this name, e.g.:

    `it('feature test', ...`

same for the others below.